### DISCUSSION TITLE
Working with huge object

### DISCUSSION THREAD

**Discussion by k11q** [pmndrs/valtio#858] [Q&A] [answered]
Hi, great project! I was thinking of something like this and came across this which is perfect and great! My use case is I was thinking of creating something like linear, where it downloads all users data, stores it in object pool, turned into object graph which then turned into observables. In the case of linear they used mobX, but I saw valtio and it seems like a good alternative for it. Do you think it can work fine say with a nested objects of 30000 items(in 1 single snapshot) to track and subscribe? Do you think there will significant performance issues?

**Comment by dai-shi** [comment, marked as answer]
All I can say is give it a try.
Valtio's `proxy()` eagerly creates nested proxified objects on initialization, so large object takes time to initialize. This is probably a unique (and intentional) behavior compared to MobX.
On the other hand, `snapshot()/useSnapshot()` creates proxies on demand, so it's fairly optimized.
